### ISSUE TITLE
Run recon from the ET ring and display plots

### ISSUE THREAD

**Issue by JeremyMcCormick** [JeffersonLab/hps-java#495]
Starter script (from hps-java-sandbox):

```
public static void main(String[] args) {
    CompositeLoopConfiguration config = new CompositeLoopConfiguration();
    LCSimEventBuilder builder = new LCSimTestRunEventBuilder();
    builder.setDebug(false);
    config.setDataSourceType(DataSourceType.ET_SERVER)
            .setDetectorName("HPS-TestRun-v8-5")
            .setEtConnection(EtConnection.createDefaultConnection())
            .setLCSimEventBuilder(builder)
            .setStopOnEndRun(true)
            .setStopOnErrors(true)
            .setSupplyLcioEvents(true)
            .setMaxQueueSize(1)
            .setTimeout(-1L)
            .add(new CalibrationDriver())
            .add(new RawTrackerHitSensorSetup())
            .add(new EventMarkerDriver())
            .add(new EcalRawConverterDriver());
        CompositeLoop loop = new CompositeLoop(config);
        loop.loop(-1);
}
```

**Comment by JeremyMcCormick** [comment]
Initially implement and merge a single thread/process command line tool.

Ideally, we would have a tool for using multiple cores but this is fairly complex.

**Comment by JeremyMcCormick** [comment]
Sent: Monday, June 3, 2019 2:07 PM
...
Subject: Re: ET ring processing

Hello Norman,

attached is our java version of the histograms for online data processing, which was written but never used since we are using c++ version, so it was not extensively tested. I compiled it and run, it books 1dim histogram and sends it over activemq, receiver see it and decode correctly.

To compile:

```
mvn install
```

To run (assuming ActiveMQ server on host clon00, and topic is 'clasrun.clastest1.daq.hist'): 

```
java -cp target/clasMQ-3.0-SNAPSHOT-jar-with-dependencies.jar org.jlab.clas.mq.IpcServer tcp://clon00.jlab.org:61616 clasrun.clastest1.daq.hist
```

Please let me know if it is usable. If so you can incorporate it into data processing, and we'll test it
here along with the rest of the system. As I described it on the meeting, main idea is to have many senders and our ROOT-based receiver which will fill and display histograms from all senders.

Will give info on ET producer tomorrow.

Sergey

**Comment by JeremyMcCormick** [comment]
Sent: Monday, June 3, 2019 2:25 PM
...
Subject: Re: ET ring processing

Hello Norman,

to insert data from evio file to ET:

1. Start ET system: et_start -n 100 -s 2000000 -f /et/test1
2. Start ET producer: evio2et  evio_file_name  /et/test1
3. Start ET monitor (just to control): et_monitor -f /et/test1

It will read file over and over again and insert data into ET.
Instead of 'evio2et' program 'evio2mem2et' can be used if memory allows, it will read entire file into memory, event rate will be higher in that case.

It all resides in $CODA/src/et-14.0/ on clon machines. Please let me know if it works for you.

NOTE: only one ET system can be run on one machine, so if you are running 'et_start' and see message:
  et SEVERE: et_netserver: specified port is busy, cannot start server thread
then search for 'et_start' or 'coda_et' processes and kill them.

Sergey

**Comment by JeremyMcCormick** [comment]
[java.zip](https://github.com/JeffersonLab/hps-java/files/3251179/java.zip)

**Comment by JeremyMcCormick** [comment]
From: Serguei Boiarinov <boiarino@jlab.org> 
Sent: Monday, June 3, 2019 2:25 PM
Subject: Re: ET ring processing

Hello Norman,

to insert data from evio file to ET:

1. Start ET system: et_start -n 100 -s 2000000 -f /et/test1
2. Start ET producer: evio2et  evio_file_name  /et/test1
3. Start ET monitor (just to control): et_monitor -f /et/test1

It will read file over and over again and insert data into ET.
Instead of 'evio2et' program 'evio2mem2et' can be used if memory allows, it will read entire file into memory, event rate will be higher in that case.

It all resides in $CODA/src/et-14.0/ on clon machines. Please let me know if it works for you.

NOTE: only one ET system can be run on one machine, so if you are running 'et_start' and see message:
  et SEVERE: et_netserver: specified port is busy, cannot start server thread
then search for 'et_start' or 'coda_et' processes and kill them.

**Comment by normangraf** [comment]
Got the following error when trying to build iss495:
[ERROR] Failed to execute goal on project hps-record-util: Could not resolve dependencies for project org.hps:hps-record-util:jar:4.4-SNAPSHOT: Could not find artifact org.jlab.coda:et:jar:16.0 in maven-central-repo (http://repo1.maven.org/maven2/) -> [Help 1]

**Comment by mholtrop** [comment]
The ET system needs to be at 16.2 which can be found here: https://coda.jlab.org/maven/org/jlab/coda/et/16.2/ <https://coda.jlab.org/maven/org/jlab/coda/et/16.2/>
The 16.0 no longer exists.

I updated the pom.xml and it should work now.

**Comment by normangraf** [comment]
Thank you, that fixes the record-util issue.

**Comment by JeremyMcCormick** [comment]
Can you check in that change if you didn't already?

**Comment by mholtrop** [comment]
It _was_ checked in. 

commit eb2154efe752458ea4c5a22c18cfd17e841c00b8 (HEAD -> iss495, origin/iss495)
Author: Maurik Holtrop <maurik@physics.unh.edu>
Date:   Fri Jun 14 11:33:18 2019 -0400

    Move ET to 16.2 which can actually be found at JLab. :-)

**Comment by JeremyMcCormick** [comment]
This is resolved by #511.

I have additional features that I would like to implement but this is a good starter version.
